api/app/src/main/kotlin/packit/service/RoleService.kt

```diff
@@ -40,26 +43,34 @@ class BaseRoleService(
             ?: throw PackitException("adminRoleNotFound", HttpStatus.INTERNAL_SERVER_ERROR)
     }
 
-    override fun saveRole(roleName: String)
+    override fun createRole(createRole: CreateRole)
+    {
+        val permissions = permissionService.checkMatchingPermissions(createRole.permissions)
+
+        saveRole(createRole.name, permissions)
+    }
+
+    internal fun saveRole(roleName: String, permissions: List<Permission> = listOf())
     {
         if (roleRepository.existsByName(roleName))
         {
             throw PackitException("roleAlreadyExists")
         }
         val role = Role(name = roleName)
+        role.rolePermissions = permissions.map { RolePermission(permission = it, role = role) }
+            .toMutableList()
```

Comment:
Could be `toList()`, this doesn't need to be mutable. Oh, except it's defined as mutable on the entity type.. maybe it's doesn't need to be mutable there either, seems a bit odd..? Depends on how you're going to do updates I guess.

Reply:
yeah i wish but cant because its a mutable list.. needs to be mutable because we will be updating the list (adding/ removing)